include/deal.II/lac/trilinos_tpetra_vector.h

```diff
@@ -664,12 +702,19 @@ namespace LinearAlgebra
        * necessary after writing into a vector element-by-element and before
        * anything else can be done on it.
        *
+       * The (defaulted) argument can be used to specify the compress mode
+       * (<code>Add</code> or <code>Insert</code>) in case the vector has not
+       * been written to since the last time this function was called. The
+       * argument is ignored if the vector has been added or written to since
+       * the last time compress() was called.
+       *
        * See
        * @ref GlossCompress "Compressing distributed objects"
        * for more information.
        */
       void
-      compress(const VectorOperation::values operation);
+      compress(const VectorOperation::values operation,
+               const IndexSet               &locally_owned = IndexSet());
```

Comment:
I reworked the description:

```
      /**
       * Compress the underlying representation of the Trilinos object, i.e.
       * flush the buffers of the vector object if it has any. This function is
       * necessary after writing into a vector element-by-element and before
       * anything else can be done on it.
       *
       * @param operator The compress mode (<code>Add</code> or <code>Insert</code>)
       * in case the vector has not been written to since the last time this
       * function was called. The argument is ignored if the vector has been
       * added or written to since the last time compress() was called.
       *
       * @param locally_owned (Optional) Provide the locally owned index set
       * corresponding to this vector.
       *
       * See
       * @ref GlossCompress "Compressing distributed objects"
       * for more information.
       */
```